src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    /// <summary>
+    /// Handles the get code actions command.
+    /// </summary>
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentCodeActionName)]
+    internal class CodeActionsHandler : CodeActionsHandlerBase, IRequestHandler<LSP.CodeActionParams, LSP.Command[]>
+    {
+        [ImportingConstructor]
+        public CodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService) : base(codeFixService, codeRefactoringService)
+        {
+        }
+
+        public async Task<LSP.Command[]> HandleRequestAsync(Solution solution, LSP.CodeActionParams request,
+            LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            var codeActions = await GetCodeActionsAsync(solution,
+                                                    request.TextDocument.Uri,
+                                                    request.Range,
+                                                    cancellationToken).ConfigureAwait(false);
+
+            // Filter out code actions with options since they'll show dialogs and we can't remote the UI and the options.
+            codeActions = codeActions.Where(c => !(c is CodeActionWithOptions));
```

Comment:
we should create work item to figure out how to support code action with options.

Reply:
@rchande is this how C# extensino on vscode work as well?

Reply:
@heejaechang More or less. There's an open PR to support a specific set of them: https://github.com/OmniSharp/omnisharp-roslyn/pull/1406

Reply:
https://github.com/dotnet/roslyn/projects/45#card-20873003